### ISSUE TITLE
Implement GAN flavours

### ISSUE THREAD

**Issue by gcastro-98** [gcastro-98/synthetic-medical-images#2]
Apart from the vanilla conditional [cGAN](https://arxiv.org/abs/1411.1784), the following seems promising for CV:
- [x] [Deep Convolutional cGAN (cDCGAN)](https://arxiv.org/abs/1511.06434): 
  - Note a DCGAN can be first [constructed](https://pytorch.org/tutorials/beginner/dcgan_faces_tutorial.html), and then [conditioned](https://machinelearningmastery.com/how-to-develop-a-conditional-generative-adversarial-network-from-scratch/) to the labels 
- [ ] [Self-attention GAN](https://arxiv.org/abs/1805.08318)
  - Note this [`tensorflow` implementation](https://github.com/brain-research/self-attention-gan) could be adapted to `pytorch`
- [ ] [SynGAN](https://arxiv.org/abs/1908.09899) 

Or we could even try other generative models such as diffusion ones:
- [ ] Class-conditional diffusion models: with this interesting [tutorial](https://github.com/Project-MONAI/GenerativeModels/blob/main/tutorials/generative/classifier_free_guidance/2d_ddpm_classifier_free_guidance_tutorial.ipynb)
